client/settings/survey-modal/index.js

```diff
@@ -0,0 +1,200 @@
+import React, { useContext, useEffect } from 'react';
+
+import { __ } from '@wordpress/i18n';
+import { dispatch } from '@wordpress/data';
+import {
+	Button,
+	Icon,
+	RadioControl,
+	TextareaControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import ConfirmationModal from 'components/confirmation-modal';
+import useIsUpeEnabled from 'settings/wcpay-upe-toggle/hook';
+import { wcPaySurveys } from './questions';
+import WcPaySurveyContext from './context';
+import { useSurveySubmit, useSurveyAnswers } from './hook';
+
+const DisabledUPESuccessNotice = () => {
+	return (
+		<div className="disable-success-notice">
+			<Icon className="disable-success-icon" icon="yes-alt" />
+			<p>
+				{ __(
+					"You've disabled the new payments experience in your store.",
+					'woocommerce-payments'
+				) }
+			</p>
+		</div>
+	);
+};
+
+const SurveyPromptQuestionDisabledUPE = () => (
+	<p>
+		<strong>
+			{ __(
+				'What made you disable the new payments experience?',
+				'woocommerce-payments'
+			) }
+		</strong>
+	</p>
+);
+
+const SurveyPromptQuestion = () => (
+	<p>
+		<strong>
+			{ __(
+				'Do you have any feedback for the new payments experience?',
+				'woocommerce-payments'
+			) }
+		</strong>
+	</p>
+);
+
+const SurveyModalBody = ( { optionsArray, surveyQuestion } ) => {
+	const [ isUpeEnabled ] = useIsUpeEnabled();
+	const [ surveyAnswers, setSurveyAnswers ] = useSurveyAnswers( {
+		surveyQuestion: '',
+		comments: '',
+	} );
+	return (
+		<>
+			{ ! isUpeEnabled ? (
+				<>
+					<DisabledUPESuccessNotice />
+					<SurveyPromptQuestionDisabledUPE />
+				</>
+			) : (
+				<SurveyPromptQuestion />
+			) }
+			<RadioControl
+				options={ optionsArray }
+				onChange={ ( value ) =>
+					setSurveyAnswers( {
+						...surveyAnswers,
+						[ surveyQuestion ]: value,
+					} )
```

Comment:
It looks like `setSurveyAnswers` leverages the previous value of `useState` to update the `surveyAnswers`.

When this pattern is used, it's best to use `setState(...)` as a callback.
In this case, it would be:
```
					setSurveyAnswers( ( prev ) => ( {
						...prev,
						[ surveyQuestion ]: value,
					} ) )
```

See https://reactjs.org/docs/react-component.html#setstate (in this case it's for `setState` on a class-based component, but the rule still applies.

Reply:
🤦 Hooks made me forget about bringing in prevState. Thanks!